src/commands/deployment/correlate.ts

```diff
@@ -76,45 +83,46 @@ export class DeploymentCorrelateCommand extends Command {
       maxConcurrentProcesses: 2, // max 2 git commands at the same time
     })
 
-    const currentBranch = await gitCurrentBranch(git)
-    if (!currentBranch) {
-      this.logger.error('Could not get current branch')
+    if (!this.configurationRepo) {
+      this.configurationRepo = await gitRepositoryURL(git)
+    }
+
+    if (this.configurationRepo === undefined || this.configurationRepo === '') {
+      this.logger.error('Could not retrieve repository URL, check out a repository or provide it with --config-repo')
 
       return 1
     }
 
-    let localCommitShas: string[]
-    if (this.configurationRepo) {
-      localCommitShas = await gitLocalCommitShas(git, currentBranch)
-    } else {
-      ;[this.configurationRepo, localCommitShas] = await Promise.all([
-        gitRepositoryURL(git),
-        gitLocalCommitShas(git, currentBranch),
-      ])
+    if (!this.configurationShas) {
+      this.logger.info('Retrieving local git commits')
+      const currentBranch = await gitCurrentBranch(git)
+      if (!currentBranch) {
+        this.logger.error('Could not get current branch')
+
+        return 1
+      }
+      this.configurationShas = await gitLocalCommitShas(git, currentBranch)
     }
 
-    if (this.configurationRepo === undefined || this.configurationRepo === '') {
-      this.logger.error('Could not retrieve repository URL, check out a repository or provide it with --config-repo')
+    if (this.configurationShas.length === 0) {
+      this.logger.error(
+        'Could not retrieve commit SHAs, make commits and then call this command or provide them with --config-shas'
```

Comment:
I think this is clearer
```suggestion
        'Could not retrieve commit SHAs, commit changes and then call this command or provide them with --config-shas'
```